_posts/2022-09-21-hvr-and-hva.markdown

```diff
@@ -0,0 +1,110 @@
+
+## HVA
+
+### Overview
+
+High Volume Agent (HVA) is the marriage of HVR’s replication capabilities with Fivetran’s ease of use. HVR was recently acquired by Fivetran, and in line with Fivetran’s objectives, HVA hopes to house its replication technology within Fivetran’s core product. Presently, HVA is an enterprise ready Fivetran connector for Oracle and MySQL. The throughput of this replication is a staggering 10+ mB/s. The major use case for this tool is for enterprise companies that produce a large amount of data or have large amounts of historical data, that wish to transport its data and leverage one of the many cloud based storage solutions Fivetran integrates with.
+
+### Issues solved by HVA
+
+- Faster and higher volume replication
+- Major compute impacts of replication
+- Complicated process of replication
+
+HVA seeks to tackle both of these issues in the data space. Data collection is ever increasing and enterprises today are racing to leverage their own data. Thus, speed and volume of replication are two metrics paramount to success in this race. Both of which are provided by HVA. With more than 10 mB/s being replicated a second, HVA is a leader in both speed and volume. In addition, the utilisation of change data capture, reading only the changes made to a database, means that the database isn’t directly queried. Similar to HVR, HVA leverages the database’s change data capture transaction logs to do this. This avoids querying the database for incremental loads, and the compute load on the database itself is minimised. Finally, HVA is a simple tool to use. In both setup and use, Fivetran relative to other products minimises the typically verbose and drawn out process of setting up a replication service. Once HVA’s agent is set up, it can easily be plugged into your Fivetran account and thus its numerous destination options.
+
+### Setup
+
+![Architecture](/img/blog/hvr-and-hva/hva-architecture.png)
+
+HVA uses an agent-based approach in which an ‘agent’ is installed on the database server. This is due to HVA’s need to read the change data capture transaction logs.
+
+The setup for an Oracle Database is relatively simple. HVA currently supports Generic Oracle and Oracle RAC services. The setup follows these high level steps:
+
+- Choose and configure the type of connection to the source database. These options include:
+  - Direct
+  - SSH tunnel
+  - AWS PrivateLink
+- Setup a read-only access User for Fivetran’s agent
+- Configure Archivelog Mode, Supplemental logging and Configure Direct - Capture to be enabled.
+- Install and configure the HVA Agent
+- Start the HVA Agent
+
+A complete setup guide can be found here: https://fivetran.com/docs/databases/oracle/oracle_hva/setup-guide
+
+Once this is complete, a connector can be added to your Fivetran account via the dashboard or dashboard.
+
+## Use Cases (based on Oracle databases)
```

Comment:
Why Oracle?

What I really want to know is what sucks about it?
And when I should use it?? (and when I shouldn't

Reply:
Maybe dont limit to just oracle, but highlight the difficulties of getting stuff of it

Depending on our position, give a example of replication to Snowflake, which will most likley be use case for MR clients

https://fivetran.com/docs/hvr6/requirements/source-and-target-requirements/snowflake-requirements/snowflake-as-target
https://fivetran.com/docs/hvr6/requirements/source-and-target-requirements/oracle-requirements/oracle-as-source

HVA can only do oracle at the moment
HVR can pretty much do anything (including file systems)
Fivetran can also do Oracle...but very limited

Reply:
Correction, HVA can also do SQL server now